logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/PersistMessageUseCase.kt

```diff
@@ -0,0 +1,49 @@
+package com.wire.kalium.logic.data.message
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.onSuccess
+
+/**
+ * Internal UseCase that should be used instead of MessageRepository.persistMessage(Message)
+ * It automatically updates ConversationModifiedDate and ConversationNotificationDate if needed
+ */
+interface PersistMessageUseCase {
+    suspend operator fun invoke(message: Message): Either<CoreFailure, Unit>
+}
+
+class PersistMessageUseCaseImpl(
```

Comment:
```suggestion
internal interface PersistMessageUseCase {
    suspend operator fun invoke(message: Message): Either<CoreFailure, Unit>
}

internal class PersistMessageUseCaseImpl(
```

Reply:
it can't be `internal` as it's used in some other packages